components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -25,23 +29,55 @@ internal class ExperimentsUpdater(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal lateinit var source: KintoExperimentSource
 
-    @Volatile private var updateJob: Job? = null
-
     internal fun initialize(configuration: Configuration) {
         source = getExperimentSource(configuration)
 
-        updateJob = GlobalScope.launch(IO) {
-            // Update the experiments list from the server async, without blocking
-            // the app launch.
-            updateExperiments()
-        }
+        // Schedule the periodic experiment updates
+        scheduleUpdates()
     }
 
-    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
-    internal suspend fun testWaitForUpdate() {
-        updateJob!!.join()
+    /**
+     * Schedule update with the default constraints, but only if the periodic worker is not already
+     * scheduled.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun scheduleUpdates() {
+        WorkManager.getInstance().enqueueUniquePeriodicWork(
+            TAG, ExistingPeriodicWorkPolicy.KEEP, getWorkRequest())
     }
 
+    /**
+     * Builds the periodic work request that will control the scheduling of the worker and provide
+     * the tag value so that the worker can be interacted with later.
+     *
+     * @return [PeriodicWorkRequest]
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun getWorkRequest(): PeriodicWorkRequest {
+        return PeriodicWorkRequest.Builder(
```

Comment:
Per our conversation, this also can trigger when in the background.
This has some implications for how experiments run, we should file a follow-up bug and talk through that.

Reply:
Follow up bug [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1544931)